fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java

```diff
@@ -112,7 +114,9 @@ public double getMemCost() {
         JoinExecMode execMode = deriveJoinExecMode();
         double rightOutput = rightStatistics.getOutputSize(context.getChildOutputColumns(1));
         double memCost;
-        int beNum = Math.max(1, ConnectContext.get().getAliveBackendNumber());
+        int beNum = Math.max(1, ConnectContext.get().getAliveBackendNumber() +
```

Comment:
would you please leave a `TODO` here, it may not be accurate in shared-data cluster using all alive compute nodes to estimate the cost, ideally it should be warehouse awareness.

Reply:
Of course. TODO added. Exactly it should involve more elements to make the cost more accurate.

Reply:
Need to be tested again😭.